src/java/org/apache/cassandra/dht/tokenallocator/TokenAllocation.java

```diff
@@ -113,10 +114,10 @@ static void addOwnership(final TokenMetadata tokenMetadata, final AbstractReplic
     {
         double size = current.size(next);
         Token representative = current.getPartitioner().midpoint(current, next);
-        for (InetAddressAndPort n : rs.calculateNaturalReplicas(representative, tokenMetadata).asEndpoints())
+        for (Replica r : rs.calculateNaturalReplicas(representative, tokenMetadata))
```

Comment:
Is this an improvement? I mean this isn't a performance sensitive spot and now we have to unwrap it twice?